### ISSUE TITLE
redrover.py

### ISSUE THREAD

**Issue by ruidazeng** [JonSteinn/Kattis-Solutions#3]
I have a very similar Pythonic solution to red rover (https://open.kattis.com/problems/redrover) by iterating over sub-lists, dynamically keeping track of the macro encoding. However, my solution failed with a "WA" message failing on the **_7/38_** test case and I just couldn't figure out why.

Is your solution able to successful pass all the test cases on Kattis?

```python
route = input()
macroed = 0

for i in range(2, int(len(route)/2) + 1):
    for n in range(len(route) - i + 2):
        # if in rest of the route
        if route[n:n+i] in (route[1:n-1] + route[n+i:]):
            macroed = i
            break

if macroed == 0:
    print(len(route))
else:
    print(len(route) - macroed + 1)
```

**Comment by JonSteinn** [comment]
[https://i.ibb.co/tbxN9hB/Screen-Shot-2019-07-26-at-10-16-48-AM.png](url)

**Comment by ruidazeng** [comment]
Thank you for letting me know. I tested a sample **`EEEE`** which should return **`4`**, but my source code returned a **`3`** instead because a logic misconception that resulted in an incorrect hard-coding.

I re-evaluated my solution again. I made two minor edits and was able to AC with a CPU time of 0.03s. All I did was I start keeping track of the number of time that the red rover uses a "macros" to replace a sub-string in order to optimize the minimum characters needed while having a correct solution.
